### PR TITLE
Implement BucketMetadata::encryption.

### PR DESCRIPTION
**Pull request by coryan**
Part of the changes for #537.

### CODE REVIEW

File: google/cloud/storage/bucket_metadata.h [hunk 1]
```diff
@@ -212,6 +254,7 @@ class BucketMetadata : private internal::CommonMetadata<BucketMetadata> {
   BucketBilling billing_;
   std::vector<CorsEntry> cors_;
   std::vector<ObjectAccessControl> default_acl_;
+  BucketEncryption encryption_;
```
Comment (houglum) [inline google/cloud/storage/bucket_metadata.h]: Note: At some point, we should consider whether non-required fields should be wrapped with `optional<>`. That refactoring probably extends beyond just `encryption` and belongs in a separate commit, so I've I filed #934 for this.

File: google/cloud/storage/bucket_metadata_test.cc [hunk 2]
```diff
@@ -86,6 +86,9 @@ BucketMetadata CreateBucketMetadataForTest() {
         },
         "etag": "AYX="
       }],
+      "encryption": {
+        "defaultKmsKeyName": "/a/key/name/should/go/here"
```
Comment (houglum) [inline google/cloud/storage/bucket_metadata_test.cc]: I know this is just dummy data, but aside from having several slashes, this isn't close to a valid name.  Can we use this value instead, here and below?

```
projects/test-project-name/locations/us-central1/keyRings/test-keyring-name/cryptoKeys/test-key-name
```

File: google/cloud/storage/bucket_metadata.h (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/bucket_metadata.h]: Ack. Thanks for creating the bug btw.

File: google/cloud/storage/bucket_metadata_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/bucket_metadata_test.cc]: Done.  Ugh, who designed those.

**Comment by coryan** [comment]
PTAL.
